### PR TITLE
Unsupported statement

### LINKED ISSUE #1510: Can't parse START TRANSACTION
Here is a quick repl experiment (from version 4.2):

```
scala> import net.sf.jsqlparser.parser.CCJSqlParserUtil
import net.sf.jsqlparser.parser.CCJSqlParserUtil

scala> CCJSqlParserUtil.parse("start transaction")
net.sf.jsqlparser.JSQLParserException: Encountered unexpected token: "start" "START"
    at line 1, column 1.

Was expecting one of:

    "("
    "ALTER"
    "CALL"
    "COMMENT"
    "COMMIT"
    "CREATE"
    "DECLARE"
    "DELETE"
    "DESCRIBE"
    "DROP"
    "EXEC"
    "EXECUTE"
    "EXPLAIN"
    "GRANT"
    "IF"
    "INSERT"
    "MERGE"
    "PURGE"
    "RENAME"
    "REPLACE"
    "RESET"
    "ROLLBACK"
    "SAVEPOINT"
    "SET"
    "SHOW"
    "TRUNCATE"
    "UPDATE"
    "UPSERT"
    "USE"
    "VALUES"
    "WITH"
    <K_SELECT>

  at net.sf.jsqlparser.parser.CCJSqlParserUtil.parseStatement(CCJSqlParserUtil.java:190)
  at net.sf.jsqlparser.parser.CCJSqlParserUtil.parse(CCJSqlParserUtil.java:63)
  at net.sf.jsqlparser.parser.CCJSqlParserUtil.parse(CCJSqlParserUtil.java:38)
  ... 34 elided

```

### PR DESCRIPTION
**Pull request by manticore-projects**
Implement `Unsupported statement`, which will return any unparsable statement or fragment as a list of tokens.

This Feature is OFF by default and  can be activated on demand:
```java
        String sqlStr = "this is an unsupported statement";

        assertSqlCanBeParsedAndDeparsed(sqlStr, true, parser -> parser.withUnsupportedStatements(true) );

        Assertions.assertThrowsExactly(JSQLParserException.class, new Executable() {
            @Override
            public void execute() throws Throwable {
                CCJSqlParserUtil.parse(sqlStr, parser -> parser.withUnsupportedStatements(false) );
            }
        });
```

However, there are 2 caveats:
1) It does not work within `BEGIN ... END` blocks (because its hard to distinguish  the `END` token from an `Unsupported Statement`
2) It is always ACTIVE for the `Statements()` production since the needed `LOOKAHEAD` does not work, but results in a stack overflow (maybe a JavaCC bug)

Also `IF ... ELSE ...` has not been considered yet. I do not think, it is relevant.

Fixes #1510

### CODE REVIEW

File: src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt [hunk 1]
```diff
@@ -655,10 +657,27 @@ Block Block() #Block : {
 }
 {
     <K_BEGIN>
-(<ST_SEMICOLON>)*
+    (<ST_SEMICOLON>)*
     try {
-       (stm = SingleStatement() | stm = Block()) { list.add(stm); } <ST_SEMICOLON>
-       ( (stm = SingleStatement() | stm = Block()) <ST_SEMICOLON> { list.add(stm); } )*
+       (
```
Comment (wumpz) [inline src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt:662]: To be honest I find your version much harder to read.

**Review by wumpz — APPROVED** [review]
However, I will merge. I find the idea of parsing without exceptions very attractive. :)

**Comment by vkostyukov** [comment]
I like this. Any idea when this is going to be released?

**Comment by wumpz** [comment]
At the moment I am deploying a new SNAPSHOT version. To be honest releasing this was not yet a plan, since version 4.4 was released recently.

**Pull request merged**